### PR TITLE
Remove panic-catching and fix tests that use newly-working native try_call

### PR DESCRIPTION
**Pull request by graydon**
fixes https://github.com/stellar/rs-soroban-env/issues/462
fixes https://github.com/stellar/rs-soroban-sdk/issues/642

This "fixes" the above bugs by removing the panic-catching machinery from the SDK as discussed with @leighmcculloch -- the host now does a `catch_panic` inside `try_call` and calls `rollback` itself, so it functions "like the VM". This change is mostly just removing stuff from the SDK and tweaking tests to expect correct behaviour.

I think the only slightly hinky thing in here is that the tests assert on a _specific_ status that differs depending on whether the panic was caught natively by the host or due to a vm trap. I think this is the best we can do here and now, but we might want to make a note of it somewhere as a consideration when assigning the final error-code repertoire: it might be better to merge these two error codes and just restrict the difference between a failed VM `try_call` and a failed native `try_call` to differences in `DebugEvent`s

cc @jayz22

### CODE REVIEW

File: soroban-sdk/src/tests/contract_invoke.rs [hunk 1]
```diff
@@ -21,9 +20,15 @@ fn test_invoke() {
 }
```
Comment (leighmcculloch) [inline soroban-sdk/src/tests/contract_invoke.rs:20]: @graydon Could we make this test pass by retaining the panic string and plopping it into a debug event?

There are plenty of examples that rely on the panic string in the test. I would like us to move away from that type of testing and instead use errors, but I don't want to take away the ability to should on the error string if we don't have to.

**Review by leighmcculloch — COMMENTED** [review]
🎉 This looks great. I have one ask inline, that I think requires a tweak to the env.

File: soroban-sdk/src/tests/contract_invoke.rs (hunk 1 above)
Comment (graydon) [reply soroban-sdk/src/tests/contract_invoke.rs:20]: Env tweaked! Posting update.

**Review by leighmcculloch — APPROVED** [review]
🎉🎉🎉🎉🎉🎉

**Comment by leighmcculloch** [comment]
> it might be better to merge these two error codes and just restrict the difference between a failed VM try_call and a failed native try_call to differences in DebugEvents

+1 I think doing something like this might be better. The more I think about I wonder if the distinction of "vm errors" vs non-vm errors really matters. It feels too much like an implementation detail anyway. Users of the host shouldn't need the distinction.

**Pull request merged**